tests/unit_tests/fn/test_fn_sid_neighbor_table.cpp

```diff
@@ -0,0 +1,45 @@
+/*
+ * GridTools
+ *
+ * Copyright (c) 2014-2022, ETH Zurich
+ * All rights reserved.
+ *
+ * Please, refer to the LICENSE file in the root directory.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <gridtools/fn/sid_neighbor_table.hpp>
+
+#include <array>
+#include <cstddef>
+#include <cstdint>
+
+#include <gtest/gtest.h>
+
+namespace gridtools::fn {
+    namespace {
+
+        using sid_neighbor_table::as_neighbor_table;
+
+        using edge_dim_t = integral_constant<int_t, 0>;
+        using edge_to_cell_dim_t = integral_constant<int_t, 1>;
+
+        TEST(sid_neighbor_table, correctness) {
+            constexpr std::size_t num_elements = 3;
+            constexpr std::size_t num_neighbors = 2;
+            std::int32_t contents[num_elements][num_neighbors] = {{0, 1}, {10, 11}, {20, 21}};
```

Comment:
nit-picking
```suggestion
            int contents[num_elements][num_neighbors] = {{0, 1}, {10, 11}, {20, 21}};
```
we don't use std::intX_t anywhere in the codebase.

Reply:
Fair enough